app/models/ladder_match.rb

```diff
@@ -0,0 +1,54 @@
+# Public: Classes used to model matches between LadderUser classes.
+# All instance methods are validations and should not be called outside of creation/saving
+# 
+# Examples
+#   LadderMatch.point_calculate(elo1, elo2)
+
+class LadderMatch < ActiveRecord::Base
+  validates :player1,   
+            presence: true
+  validates :player2,
+            presence: true
+  validates :winner, 
+            inclusion: { in: [ 1 , 2 ], message: "Choose a winner."}
+  validates :date_of_match, presence: true
+    
+  #custom methods
+  validate :check_user_registered1, :check_user_registered2, :check_different_players, :check_match_quota,  :check_date_is_past
+    
+  #virtual attribute
+  attr_accessor :password
+    
+  #All the custom validators - are users registered, are users different players, have they played the maximum number of games, is the date physically possible
+  def check_user_registered1
+    if (LadderUser.find_by username: player1).blank?
+      errors.add(:player1, "Player 1 is not registered")
+    end
+  end
+  def check_user_registered2
+    if (LadderUser.find_by username: player2).blank?
+      errors.add(:player2, "Player 2 is not registered")
+    end
+  end
+  def check_different_players
+    if (player1 === player2)
+      errors.add(:player2, "You can't play yourself!")
+    end
+  end
+  def check_match_quota
+    matches_played = LadderMatch.where("player1 = ? AND player2 = ?", player1, player2).length + LadderMatch.where("player1 = ? AND player2 = ?", player2, player1).length
```

Comment:
In the match instructions, I say that you need to be entered by only one player. I'm not sure if there's any other better way to do it, since players should probably be allowed to play multiple matches in one day.